Makefile

```diff
@@ -270,12 +270,15 @@ publish-elyra-image: elyra-image # Publish Elyra stand-alone container image
 	# this is a privileged operation; a `docker login` might be required
 	docker push docker.io/$(ELYRA_IMAGE)
 	docker push quay.io/$(ELYRA_IMAGE)
-	# If we're building a release from main, tag latest and push
-	if [ "$(TAG)" != "dev" -a "$(shell git branch --show-current)" == "main" ]; then \
-		docker tag docker.io/$(ELYRA_IMAGE) docker.io/$(ELYRA_IMAGE_LATEST); \
-		docker push docker.io/$(ELYRA_IMAGE_LATEST); \
-		docker tag quay.io/$(ELYRA_IMAGE) quay.io/$(ELYRA_IMAGE_LATEST); \
-		docker push quay.io/$(ELYRA_IMAGE_LATEST); \
+	# If we're building an official release, tag latest and push
+	if [ "$(TAG)" != "dev" ]; then \
+		if [ "$(shell git branch --show-current)" == "main" ] || \
+		   [ "$(shell git describe --tag)" == "v${TAG}" ]; then \
```

Comment:
If the tag corresponds to a patch release of an older minor release (e.g., `3.4.1` when `3.10` is the latest minor), won't this push `latest` images that correspond to `3.4.1`?

I suppose pre-releases could be handled similarly since I think we'd want _latest_ to equate to the _latest official release_.

Perhaps we need an indicator that this is a "latest" release vs. a "hotfix" or "pre-release" or something to that effect?

Reply:
I've looked into this some more, with the intent to automate the 'needs-to-be-tagged-as-latest' task to prevent user errors. Both registries we are publishing to provide an API that can be used to query tags, so a basic semver-based analysis _should_ be able to make the determination whether a release is the `latest` or not:

Example API request for Docker Hub for the `elyra` repository:
```
curl https://registry.hub.docker.com/v1/repositories/elyra/elyra/tags
```

Example API request for quay.io for the `elyra` repository:
```
curl https://quay.io/api/v1/repository/elyra/elyra/tag/
```

Reply:
Hmm.  Given the curl against the docker registry timed out and the one against quay.io returns LOTS of data, I think it would be less error-prone to somehow make this determination from the build env.

I was thinking the images were built from the release script and you could drive this boolean value from flags like `--rc`, `--beta`, and perhaps a new `--field-patch` (implying a patch to a non-latest version).

Since all non-dev images must be built from a tag (when would they ever be built on `main`?), it seems like you could gather the tags, sort them via semver (much easier said than done from a Makefile), and check if the $TAG matches the first (most current) tag in the set.

Seems like an option to `create-release.sh` that can deal with the sort logic would be ideal.  This option could be a special "build-images" option since you usually want to check the release out first, but could also enable an "all-inclusive" option when building the release in general.

Reply:
I'm not sure that extra level of indirection is necessary because the hard-wired version string in the `Makefile` already [explicitly includes information that identifies a release as beta or candidate](https://github.com/elyra-ai/elyra/blob/main/create-release.py#L841-L844). To test ^^ out I wrote a small Python script that queries the dockerhub|quay registry and returns the largest official version number as a string, e.g. "3.10.1". The Makefile could evaluate that value (similar to what [we already do for the branch](https://github.com/elyra-ai/elyra/blob/main/Makefile#L274)) to determine whether the `latest` tag needs to be applied or not. 

I do agree that the check for the branch name can/should be removed, if building a release from a tag is the only supported option.

Reply:
> I'm not sure that extra level of indirection is necessary because the hard-wired version string in the Makefile already [explicitly includes information that identifies a release as beta or candidate](https://github.com/elyra-ai/elyra/blob/main/create-release.py#L841-L844).

Yes, you could interrogate the `TAG` or `VERSION` indicator from the Makefile looking for '`b`' and '`rc`', but this still wouldn't help with field patches.   I'm just saying it would be simpler to introduce another flag `IS_LATEST`(?) 
 that the release script produces based on an already required CLI option and then the Makefile just use that.  This new flag could then be something users of the Makefile could also set to produce images outside of the release script.

Looking at the release script, I think you'd have something like:

is_latest = (`git_branch` == 'main' and not `rc` and not `beta`)

(Personally, I'd create another field in the configuration structure called `pre_release` which is a boolean of (`rc` or `beta`).  Then you'd have IS_LATEST = (`git_branch` == 'main' and not `pre_release`) and invoke `make IS_LATEST=f"{is_latest}" publish-images`.)

Reply:
> Yes, you could interrogate the `TAG` or `VERSION` indicator from the Makefile looking for '`b`' and '`rc`', but this still wouldn't help with field patches.

Maybe we can simplify things and just add a prompt in the release script asking the user, if this is a maintenance release or not (Y/N) and then update as necessary?

Reply:
This would have some issues in scripting environments (or we'd need to use a `-y` option) - which I think is one of the things @ptitzler is focused on.  In my opinion, and stated previously, adding a new option (perhaps `--maint`?) is the easiest way to go.  (But I also like the auto-detection approach listed previously. :smile: )